--- libtirpc-0.1.7/doc/etc_netconfig.orig	2005-05-18 01:10:50.000000000 -0400
+++ libtirpc-0.1.7/doc/etc_netconfig	2006-08-11 11:06:06.000000000 -0400
@@ -10,8 +10,8 @@
 # The <device> and <nametoaddr_libs> fields are always empty in this
 # implementation.
 #
-#udp6       tpi_clts      v     inet6    udp     -       -
-#tcp6       tpi_cots_ord  v     inet6    tcp     -       -
+udp6       tpi_clts      v     inet6    udp     -       -
+tcp6       tpi_cots_ord  v     inet6    tcp     -       -
 udp        tpi_clts      v     inet     udp     -       -
 tcp        tpi_cots_ord  v     inet     tcp     -       -
 rawip      tpi_raw       -     inet      -      -       -
