diff --git a/src/clnt_raw.c b/src/clnt_raw.c
index 153dd87..36035c8 100644
--- a/src/clnt_raw.c
+++ b/src/clnt_raw.c
@@ -82,12 +82,13 @@ clnt_raw_create(prog, vers)
 	rpcprog_t prog;
 	rpcvers_t vers;
 {
-	struct clntraw_private *clp = clntraw_private;
+	struct clntraw_private *clp;
 	struct rpc_msg call_msg;
 	XDR *xdrs = &clp->xdr_stream;
 	CLIENT	*client = &clp->client_object;
 
 	mutex_lock(&clntraw_lock);
+	clp = clntraw_private;
 	if (clp == NULL) {
 		clp = (struct clntraw_private *)calloc(1, sizeof (*clp));
 		if (clp == NULL) {
