--- libtirpc-1.0.1/src/Makefile.am.orig	2016-08-31 20:47:28.787746216 +0200
+++ libtirpc-1.0.1/src/Makefile.am	2016-08-31 20:50:52.087743898 +0200
@@ -12,8 +12,9 @@
 
 lib_LTLIBRARIES = libtirpc.la
 
-libtirpc_la_LDFLAGS = @LDFLAG_NOUNDEFINED@ -no-undefined -lpthread
+libtirpc_la_LDFLAGS = @LDFLAG_NOUNDEFINED@ -no-undefined
 libtirpc_la_LDFLAGS += -version-info @LT_VERSION_INFO@
+libtirpc_la_LIBADD = -lpthread
 
 libtirpc_la_SOURCES = auth_none.c auth_unix.c authunix_prot.c bindresvport.c clnt_bcast.c \
         clnt_dg.c clnt_generic.c clnt_perror.c clnt_raw.c clnt_simple.c \
